src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -24,35 +27,41 @@ public OrasClient(string imageReference)
             _imageReference = imageReference;
         }
 
-        public async Task PullImageAsync(string outputFolder)
+        public async Task<string> PullImageAsync(string outputFolder)
         {
             string command = $"pull  \"{_imageReference}\" -o \"{outputFolder}\"";
-            await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string output = await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            Console.WriteLine(output);
+            var digest = _digestRx.Matches(output)[0].Groups["digest"].ToString();
+            return digest;
         }
 
-        public async Task PushImageAsync(string inputFolder)
+        public async Task PushImageAsync(string inputFolder, List<string> filePathList)
         {
             string argument = string.Empty;
             string command = $"push \"{_imageReference}\"";
 
-            var filePathToPush = Directory.EnumerateFiles(inputFolder, "*.tar.gz", SearchOption.AllDirectories);
-
             // In order to remove image's directory prefix. (e.g. "layers/layer1.tar.gz" --> "layer1.tar.gz"
             // Change oras working folder to inputFolder
-            foreach (var filePath in filePathToPush)
+            foreach (var filePath in filePathList)
             {
-                argument += $" \"{Path.GetRelativePath(inputFolder, filePath)}\"";
+                if (!File.Exists(Path.Combine(inputFolder, filePath)))
+                {
+                    throw new OverlayException(TemplateManagementErrorCode.ImageLayersNotFound, "Image layer not found");
+                }
+
+                argument += $" \"{filePath}\"";
```

Comment:
What if we pass some files that are not exist to oras cli? Seems the files added to the argument have chance to be deleted before line 61